core/services/AuthService.js

```diff
@@ -58,12 +58,45 @@ module.exports = {
     return _.omit(result, "iat")
   },
 
-  checkRight: (operator, modelKey, right) =>
-    !!(operator.isAdministrator
-      || (
-        operator.rights
-        && operator.rights[modelKey.toLowerCase()]
-        && operator.rights[modelKey.toLowerCase()].includes(right.toLowerCase())
-      )
-    ),
+  /**
+   * Check if operator has rights to perform an action
+   * @param {Object} operator The operator object
+   * @param {String} modelKey The name of the model for which we are checking rights
+   * @param {String} right The name of the right
+   * @returns {Promise<boolean>} True if operator has rights and false if not
+   */
+  checkRight: async (operator, modelKey, right) => {
+    if (operator.isAdministrator) {
+      return true
+    }
+
+    if (operator.rights
+      && operator.rights[modelKey.toLowerCase()]
+      && (!(operator.rights[modelKey.toLowerCase()][right.toLowerCase()] == null))
```

Comment:
This is good that you compare it with non-strict equals `==`, so it will work for `null` and `undefined` both. But, this is still not very good for code style. It's better to use special function `_.isNil` from lodash.